core/string.rbs

```diff
@@ -897,10 +897,10 @@ class String
   #     number #=> "9"
   #
   def =~: (Regexp regex) -> Integer?
-        | [T] (_MatchAgainst[self, T] object) -> T
+        | [T] (_MatchAgainst[T] object) -> T
 
-  interface _MatchAgainst[O, T]
```

Comment:
It didnt make much sense to have an `O` field here, as it's only ever being matched against `String`s, so I've removed the field. However, it is a breaking change, so we might want to hold off on this

Reply:
Changing from `O = self` to `String` loses type narrowing.

Having a generic `#=~` (with both LHS and RHS type variables) is also fine for #1638.

Reply:
Resolved by reverting this